cibyl/orchestrator.py

```diff
@@ -294,19 +328,34 @@ def extend_parser(self, attributes: dict, group_name: str = 'Environment',
             if has_api:
                 # API entry is related to a model that has an API
                 new_group_name = class_type.__name__
+                next_parent_queries = parent_queries
                 if arguments:
-                    # add the arguments found in the current entry, but relate
-                    # them to the model
+                    # add the arguments found in the current entry, but
+                    # group them to the model they relate to
                     self.parser.extend(arguments, new_group_name,
-                                       level=level+1)
+                                       level=level+1,
+                                       parent_queries=parent_queries)
+
+                    # generate a set of all query method associated with
+                    # the arguments
+                    query_methods = {arg.func for arg in arguments
+                                     if arg.func is not None}
+                    if query_methods:
+                        # if we found some query method in the arguments,
+                        # set it up as the parent_func to use when
+                        # recursing to explore the next Model's API
+                        next_parent_queries = query_methods
```

Comment:
What does this variable do? I only see it being assigned, but not the effect that it is to have on the next iteration.

Reply:
This is passed to the next recursive call of the `extend_parser` method, when it explores the API of another model, like for example when from the Job API we find the reference to Build and want to explore the Build API (see line 352)

Reply:
Oh, I see